mlos_bench/mlos_bench/storage/base_storage.py

```diff
@@ -308,6 +308,36 @@ def new_trial(self, tunables: TunableGroups, ts_start: Optional[datetime] = None
                 An object that allows to update the storage with
                 the results of the experiment trial run.
             """
+            # Check that `config` is json serializable (e.g., no callables)
+            if config:
+                try:
+                    _config = DictTemplater(config).expand_vars()
```

Comment:
Quick hack - makes use of the fact that DictTemplater only allows primitive types throughout a nested dict structure, though we don't actually care about expanding anything.